### PR TITLE
added message counter for proposals

### LINKED ISSUE #1379: Add message counter for proposals
image
For the new design we need to show message count for each proposal card. We need to add that counter in the firebase.

### PR DESCRIPTION
**Pull request by noamso1**

### CODE REVIEW

**Comment by noamso1** [comment]
@ariel-dao This PR is done, however, we need to add the migration that counts proposal messages for the first time, for existing proposals.

**Comment by noamso1** [comment]
Done. Now we're gonna have proposal.messageCount. I also ran the migration to retroactively populate the values in all 3 environments.

File: functions/src/discussionMessage/triggers.ts [hunk 1]
```diff
@@ -32,14 +34,27 @@ exports.watchForDeletedMessages = functions.firestore
...
   let commonId : string
+  
   try {
     const discussion: IDiscussion = await discussionDb.getDiscussion( discussionId )
     commonId = discussion.commonId
     discussion.messageCount = num(discussion.messageCount) + v
     await discussionDb.updateDiscussion( discussionId, discussion )
   } catch {
-    logger.error("Message is not connected to a discussion.", { discussionMessage } );
+    // probably connected to a proposal and not a discussion
```
Comment (ariel-dao) [inline functions/src/discussionMessage/triggers.ts:44]: probably?

**Review by ariel-dao — COMMENTED** [review]
what is the rule of Discussions if there are messages under proposal?

File: functions/src/discussionMessage/triggers.ts (hunk 1 above)
Comment (noamso1) [reply functions/src/discussionMessage/triggers.ts:44]: Yes !! It's not an error. discussionMessage.discussionId can belong either to a discussion or to a proposal. that's the database design.

**Review by ariel-dao — APPROVED** [review]

**Pull request merged**